contracts/HypERC20.sol

```diff
@@ -11,6 +11,12 @@ import {ERC20Upgradeable} from "@openzeppelin/contracts-upgradeable/token/ERC20/
  * @dev Supply on each chain is not constant but the aggregate supply across all chains is.
  */
 contract HypERC20 is ERC20Upgradeable, TokenRouter {
+    uint8 private immutable decimalsConfig;
+
+    constructor(uint8 _decimals) {
```

Comment:
This is not even a nit, but i would love to get to a point where we can use proxies for HypErc20 since they are so heavy. Maybe we just change this then ...